src/adapters/scene_direction.js

```diff
@@ -53,6 +53,34 @@ export default class SceneDirection {
     listen('unhighlight_step', step => {
       this.unhighlightStep(step);
     });
+
+    listen('set_origin', pos => {
+      this.setOrigin(pos);
+    });
+
+    listen('set_destination', pos => {
+      this.setDestination(pos);
+    });
+  }
+
+  setOrigin = pos => {
+    const originMarker = createMarker(pos, 'itinerary_marker_origin', { draggable: true })
+      .addTo(this.map)
+      .on('dragend', event => this.refreshDirection('origin', event.target.getLngLat()));
+    this.routeMarkers.push(originMarker);
+    window.map.mb.flyTo({ center: pos });
```

Comment:
This `flyTo` ignores the paddings used by the application (on mobile for example, the marker will not be centered because of the direction panel), and uses a potentially slow animation depending on the start position.
I would suggest to reuse the functions defined in scene.js